src/dodal/devices/thawer.py

```diff
@@ -1,15 +1,53 @@
+from asyncio import Task, create_task, sleep
 from enum import Enum
 
-from ophyd_async.core import StandardReadable
+from bluesky.protocols import Stoppable
+from ophyd_async.core import (
+    AsyncStatus,
+    Device,
+    SignalRW,
+    StandardReadable,
+)
 from ophyd_async.epics.signal import epics_signal_rw
 
 
+class ThawingException(Exception):
+    pass
+
+
 class ThawerStates(str, Enum):
     OFF = "Off"
     ON = "On"
 
 
-class Thawer(StandardReadable):
+class ThawingTimer(Device):
+    def __init__(self, control_signal: SignalRW[ThawerStates]) -> None:
+        super().__init__("thawing_timer")
```

Comment:
Ok, this is an odd one. When making a `Device` super must go at the end of the function because it will iterate through and change the names of the child signals to be `device-signal`. We probably want `control` to be given it's device name from `Thawer` rather than `ThawingTimer`, in which case the answer here is for `ThawingTimer` to not be a `Device` at all, which I have done and so now it doesn't need to call super at all

Reply:
Hmm, I think that `ThawingTimer` should be a `SignalRW` with a `SoftSignalBackend`. Reading it gives you the the amount of time left for thawing (or 0), writing it does the normal `set`
Or just make it a `SignalR` and don't worry about the read

Reply:
Deleted my previous comment as I don't think it's right. I would go back to what you had before, with `__init__` before the `self._control_signal = control_signal`. I can't think of any other way which makes sense

Reply:
That still doesn't work because `ThawingTimer` overwrites the name. But I might do that anyway and spin into a new issue to think of a better way of doing it.

Reply:
Yeah, I can't think of a good solution here. You could do
```
    def __init__(self, prefix: str, name: str = "") -> None:
        self.thaw_for_time_s = ThawingTimer(epics_signal_rw(ThawerStates, prefix + ":CTRL"))
        self.control = epics_signal_rw(ThawerStates, prefix + ":CTRL")
        super().__init__(name)
```
So you'd have two separate control signals pointing to the same PV, and the names would reflect where they're from. But I can imagine creating two signals to the same PV might cause issues

Reply:
https://github.com/bluesky/ophyd-async/issues/410

Reply:
I've proposed a fix over in that ticket, but I was wondering why `ThawingTimer` is a separate object here? Could you move the `set` method into the `Thawer` and dispense with the `ThawingTimer` completely? @olliesilvester suggested they both needed to be `set` in https://github.com/bluesky/ophyd-async/issues/410#issuecomment-2189153715 but I don't see the other `set` method here...

Reply:
There isn't right now, but if there was one, it would just be setting the control signal to on or off, (no timer)

Reply:
As @olliesilvester said, the `Thawer` has two modes of operation:
1. The plan will turn it on/off
2. The plan will turn it on via the timer then not turn it off itself

I feel like having the `set` on the `Thawer` doing number 2 is really not obvious. You would logically expect it to be doing number 1.